scripts/apps/authoring-react/widget-persistance-hoc.tsx

```diff
@@ -0,0 +1,51 @@
+import {closedOnRender} from 'apps/authoring/widgets/widgets';
+import React, {RefObject} from 'react';
+import {IArticleSideWidgetComponentType} from 'superdesk-api';
+
+export let widgetState = {};
+
+interface IProps {
+    children: (
+        widgetRef: RefObject<React.PureComponent<IArticleSideWidgetComponentType>>,
+    ) => JSX.Element | Array<JSX.Element>;
+    sideWidgetId: string;
+}
+
+interface IState {
+    mounted: boolean;
+}
+
+// Functionality to persist widget state between re-renders. Solves issues related to widget state being lost when switching between different authoring views.
```

Comment:
> Solves issues related to widget state being lost

You've spent a lot of time fixing well and know it well at the moment. Could you list which issues exactly used to occur, and switching between which views?

I think it's only full-width switching? Or also widgets?